homeassistant/components/tellduslive/__init__.py

```diff
@@ -71,33 +70,30 @@
 
     hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
     hass.data[CONFIG_ENTRY_IS_SETUP] = set()
-
-    client = TelldusLiveClient(hass, entry, session)
-    hass.data[DOMAIN] = client
-    await async_add_hubs(hass, client, entry.entry_id)
-    hass.async_create_task(client.update())
-
-    interval = timedelta(seconds=entry.data[KEY_SCAN_INTERVAL])
-    _LOGGER.debug('Update interval %s', interval)
-    hass.data[INTERVAL_TRACKER] = async_track_time_interval(
-        hass, client.update, interval)
+    hass.data[INTERVAL_TRACKER] = hass.loop.create_task(
```

Comment:
We have to cancel this task with the correct task method. We can't cancel it the same way as with home assistant listeners, by just calling it. Use `task.cancel` method. But also first check that it's not done. Here's the relevant docs:
https://docs.python.org/3/library/asyncio-task.html#task-object